dblookupext/historyRepository.go

```diff
@@ -156,9 +167,21 @@ func (hr *historyRepository) RecordBlock(
 		return err
 	}
 
+	err = hr.putRoundByNonce(blockHeader)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (hr *historyRepository) putRoundByNonce(header data.HeaderHandler) error {
+	roundToByteSlice := hr.uint64ByteSliceConverter.ToByteSlice(header.GetRound())
+	nonceToByteSlice := hr.uint64ByteSliceConverter.ToByteSlice(header.GetNonce())
```

Comment:
If we record only the round-nonce pairs, then we will 3 DB accesses to fetch a block (one for retrieving the nonce, one for retrieving the hash from the nonce and the later for retrieving the block from the hash). 
Wondering if we were to record directly the round-hash we would needed only 2 DB accesses.

Reply:
Applied your suggestion and now we save block's hash by round, therefore needing only 2 DB accesses, instead of 3.